### PR TITLE
Rasterize all pages

### PR DESCRIPTION
**Pull request by henrixapp**
Rasterizes all pages of a pdf and combines them all together.

Known bugs: File is rendered in Grayscale due to dependency on https://github.com/brunsgaard/go-pdfium-render

### CODE REVIEW

File: exam_marker/Dockerfile [hunk 1]
```diff
@@ -1,5 +1,28 @@
-FROM golang:1.17
-MAINTAINER Christian Heusel <christian@heusel.eu>
+FROM golang:1.17.6-bullseye
```
Comment (christian-heusel) [inline exam_marker/Dockerfile:1]: Is there a reason to go for a fixed go version? I would rather use the `1.17-bullseye` tag!

File: exam_marker/Dockerfile [hunk 2]
```diff
@@ -1,5 +1,28 @@
-FROM golang:1.17
-MAINTAINER Christian Heusel <christian@heusel.eu>
```
Comment (christian-heusel) [inline exam_marker/Dockerfile:2]: Please re-add the `MAINTAINER` Statement

File: exam_marker/Dockerfile [hunk 3]
```diff
@@ -1,5 +1,28 @@
...
+FROM golang:1.17.6-bullseye
+# Specify pdfium version
+ARG PdfiumVersion=4861
+RUN DEBIAN_FRONTEND=noninteractive TZ=Etc/UTC apt-get -yqq update
+RUN DEBIAN_FRONTEND=noninteractive TZ=Etc/UTC apt-get -y install tzdata
+# Install pkg-config, etc.
+RUN  apt-get clean && apt-get install -yqq apt-utils pkg-config  wget
```
Comment (christian-heusel) [inline exam_marker/Dockerfile:7]: The first two lines look like they can be omitted, or is `tzdata` really needed to do other installations?
Also please make this a single `RUN` statement in any case

File: exam_marker/Dockerfile [hunk 4]
```diff
@@ -1,5 +1,28 @@
...
+
+# Create .pc file for pkg-config
+RUN echo "\n" \
+     "prefix=/home\n" \
+     "Name: pdfium\n" \
+     "Description: pdfium\n" \
+     "Version: $PdfiumVersion\n" \
+     "Requires:\n" \
+     "Libs: -L/home/lib -lpdfium\n" \
+     "Cflags: -I/home/include\n" >  /home/pdfium.pc
```
Comment (christian-heusel) [inline exam_marker/Dockerfile:17]: Since this content is static anyway it could also be in a file

File: exam_marker/Dockerfile [hunk 5]
```diff
@@ -1,5 +1,28 @@
...
+
+# Download and extract pdfium binary
+RUN cd /home && wget --quiet  https://github.com/bblanchon/pdfium-binaries/releases/download/chromium%2F4861/pdfium-linux-x64.tgz \
+&& tar -xf pdfium-linux-x64.tgz && rm pdfium-linux-x64.tgz
+
+# Setting up paths for pkg-config
+ENV LD_LIBRARY_PATH=/home/lib
+ENV PKG_CONFIG_PATH=/home/
```
Comment (christian-heusel) [inline exam_marker/Dockerfile:25]: I would rather not have this in `/home` but in `/go/src/extern` or similar

File: exam_marker/Dockerfile [hunk 6]
```diff
@@ -11,6 +34,7 @@ COPY server/lti_utils ./server/lti_utils
 
 COPY exam_marker/exam_marker.go ./exam_marker/exam_marker.go
 COPY utils ./utils
-
+RUN go get github.com/brunsgaard/go-pdfium-render
```
Comment (christian-heusel) [inline exam_marker/Dockerfile:37]: Why does this have to be a separate `go get` command? If its possible please add it to the `go.mod`!

File: exam_marker/exam_marker.go [hunk 7]
```diff
@@ -108,8 +109,20 @@ func applyWatermark(input io.ReadSeeker, output io.Writer, textLeft string, text
 			return err
 		}
 	}
-
-	io.Copy(output, &tempout)
+	bits := tempout.Bytes()
+	doc, err := render.NewDocument(&bits)
+	pagesbuf := make([]io.Reader, 0)
```
Comment (christian-heusel) [inline exam_marker/exam_marker.go:114]: isnt this more like a `pageReaders` thing? :D

**Review by christian-heusel — CHANGES_REQUESTED** [review]
Just some cleanup work to do, I have yet to execute the code 😄

**Comment by christian-heusel** [comment]
I tested it now for a little, its sad that we only get grayscale output, why is that the case?

**Comment by christian-heusel** [comment]
Also please have a look at the LGTM alert!

File: exam_marker/Dockerfile (hunk 2 above)
Comment (henrixapp) [reply exam_marker/Dockerfile:2]: Changed for appropriate Label statement

File: exam_marker/Dockerfile (hunk 3 above)
Comment (henrixapp) [reply exam_marker/Dockerfile]: :heavy_check_mark:

**Comment by henrixapp** [comment]
It is gray because of line https://github.com/brunsgaard/go-pdfium-render/blob/b01df5200e992514fc6f11499bf82fbac74c8778/render.go#L105 . Probably we should rework the C interfacing.

File: exam_marker/Dockerfile (hunk 4 above)
Comment (christian-heusel) [reply exam_marker/Dockerfile]: Sorry, I overlooked `$PdfiumVersion` ... 😁

**Review by christian-heusel — APPROVED** [review]

**Pull request merged**